lib/sinatra/base.rb

```diff
@@ -1709,7 +1709,8 @@ def setup_sessions(builder)
         options = {}
         options[:secret] = session_secret if session_secret?
         options.merge! sessions.to_hash if sessions.respond_to? :to_hash
-        builder.use Rack::Session::Cookie, options
+        session_store = options.delete(:session_store) { Rack::Session::Cookie }
```

Comment:
I don't think I see an issue with it, since the original sessions hash is preserved.